index.js

```diff
@@ -66,53 +82,44 @@ class KoaWebMonetization {
     this.buckets.set(id, balance - price)
   }
 
-  paid ({ price, awaitBalance = false }) {
-    return async (ctx, next) => {
-      const id = ctx.params.id
-      if (!id) {
-        return ctx.throw(400, 'ctx.params.id must be defined')
-      }
-
-      const _price = (typeof price === 'function')
-        ? Number(price(ctx))
-        : Number(price)
-
-      if (awaitBalance) {
-        await this.awaitBalance(id, _price)
-      }
+  async receive (ctx) {
+    await this.connect()
 
-      try {
-        this.spend(id, _price)
-        return next()
-      } catch (e) {
-        return ctx.throw(402, e.message)
-      }
+    if (ctx.get('Accept').indexOf('application/spsp+json') === -1) {
```

Comment:
If we check the path in here as well and return `next()` when the path doesn't match the receiver URL and accept type, then the user could just put the receiver middleware under `app.use()` instead of needing to add a route themselves